### DISCUSSION TITLE
Code Assistance

### DISCUSSION THREAD

**Discussion by martsokha** [evcxr/evcxr#283] [Q&A] [answered]
Hey, I'm using this project as a reference as I'm trying to implement the Jupyter driver for another language in Rust.
I was wondering what this `2` means [here](https://github.com/evcxr/evcxr/blob/b7f5f9b80bb77b77eb9e04eec55a6a61cf4b2862/evcxr_jupyter/src/jupyter_message.rs#L43).

```rust
 let jparts: Vec<_> = parts.drain(delimiter_index + 2..).collect();
```

**Comment by davidlattimore** [comment, marked as answer]
If I put `dbg!(&multipart);` at the start of that function, we can see that the parts of the multipart look something like this (hashes changed slightly):

```
        b"a14c7126d1c2242f8f879020344c3b5b",
        b"<IDS|MSG>",
        b"9dadfb72cdc967c889d3cf503def489d6be8c5aac1deac1c6526481f0ce0dcdd",
        b"{\"date\": \"2023-03-18T21:31:23.580Z\", \"msg_id\": \"b164125ef8c444298a6a5ea79041fd4c\", \"username\": \"username\", \"session\": \"a14c7126d1c2242f8f879020344c3b5b\", \"msg_type\": \"kernel_info_request\", \"version\": \"5.2\"}",
        b"{}",
        b"{}",
        b"{}",
```

So it looks like we're finding the index of `<IDS|MSG>` then skipping it and the one after it and grabbing from the one that contains json onwards.

**Reply by martsokha** [reply]
Thanks for the explanation!
